### ISSUE TITLE
Mobs not pathing around larger obsticles / not wallhacking when no path exists

### ISSUE THREAD

**Issue by TeoTwawki** [DarkstarProject/darkstar#3930]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:**
30170601_0

**_Server Version_** (type `@revision` in game) **:**
e5eaf0e960992599237f7d003fac38c9811834d2

**_Source Branch_** (master/stable) **:**
master (whasf's server)

**_Additional Information_** (Steps to reproduce/Expected behavior) **:**
A veritable ton of locations are exploitable to place monster sin a situation they cannot fight back from. After a few seconds they deaggro and declaim (which..there is no reason for) but its easy to reclaim them before they get out of range.

Exmamples include 
- the lake at the south end in West Ronfaure - mobs will neither cross it nor go around it
- the ledges in La Thiene Plateau
- the lake near X:39 Y:21 Z:99 in Thiene Plateau...There is a small corner at the lake edge, where you can position a mob literally right in front of you but it can't "see" you, will face you direction and just stand there as you bash its face in with spells WS and JA. One the client end you can't melee it either.

I recall a time when In these situations mob would just wallhack through - it was thought this might be why Maat was falling beneath the BCNM when pulled back. I don't know if that got disabled, but in many of these situations the mob should have been able to take 2 steps to its left and go around the spot it could not pass through anyway!

![teo_2017 07 05_211246](https://user-images.githubusercontent.com/6871475/27891726-6b84a5ba-61c9-11e7-8a0b-3e2104e75f09.png)

![teo_2017 07 05_205642](https://user-images.githubusercontent.com/6871475/27891744-82df9c74-61c9-11e7-94ee-7a3b8ca7e98b.png)

**Comment by Jessland86** [comment]
Same thing with probably all ledges too. Like in Dragon's Aery you can skill up enfeebling on darters in the pit and they can't get to you if you're on the ledge. You can also kill Fafhogg this way too if he is close enough to the ledge when you claim him. They have to be close or it'll just draw in. But if they are close you can just cast on them all day and they can't get to you cause of the ledge. Most zones have places like this. Gusgen mines can cast on the skellies from the stairs to the outer platforms and they just stand there and take it.

**Comment by gedads** [comment]
i uploaded all files related to navmesh long ago, with reisenjima missing i think since i wasn't able to get good walls with the bamboos. They're free to download and improve if needed (can be edited with ease all are cleaned already), all .nav use settings defined by lautan, they can be improved a lot in recast navigation i use custom navmeshes which are just better but takes more space. The thing is  i have no idea how much those bigger .nav could impact a server with a lot of people

**Comment by TeoTwawki** [comment]
I don't think its just a mesh problem.

**Comment by gedads** [comment]
yes but it can be a part of it. i'm just saying that it was working better for me before the wallhack thing

**Comment by takhlaq** [comment]
if we're gonna do reduce the tile/cell size we might have reduce the search extents also or navmeshing would hit performance too hard

**Comment by TeoTwawki** [comment]
¯\\\_(ツ)\_/¯

**Comment by Cloudef** [comment]
If you go to "Ghelsba Outpost" and try to fight one of the warmachines, they won't come out of their hole. (literally)

**Comment by gedads** [comment]
either increase height or slope in recast or it may be better to just add an access ramp to the obj file in blender

**Comment by gedads** [comment]
uploaded a better map for ghelsba outpost. I 'm on ronfaure but crabs seems to turn correctly around the pond for me. redoing it anyway since it was the map lautan did in first so it's maybe not perfect.
For dragon's aery, probably adding a ramp too is enough to let the darters kill you.

**Comment by Cloudef** [comment]
Yep the gedad's updated map works for Ghelsba (apart from the tanks venturing out of their holes a bit, but that's not huge problem and probably better fixed by logic rather than navmesh).

**Comment by gedads** [comment]
"Gusgen mines can cast on the skellies from the stairs to the outer platforms" which platform?

**Comment by Jessland86** [comment]
These are the two where you can cast off the sides and mobs can't get to you.. [28 -56 -86] [28 -56 -86]

There are numerous spots where mobs don't have room to path around those end of railway stops. [20 -59 -121] [77 -59 -20] [-37 -59 -19] [-19 -59 -117] [59 -59 -122] [-20 -39 82] [60 -39 77] [220 -39 202] [77 -39 180] [-37 -39 180] [19 -39 82]

Mobs can't walk up these stairs. [9 -40 -70] [10 -20 128]

**Comment by Jessland86** [comment]
In Zeruhn Mines at [-60 7 -271] there is a rock people regularly abuse to fight high level mobs from behind that the mob can't get to you at.
